javascript/features/announce/announce_manager.js

```diff
@@ -62,18 +66,42 @@ class AnnounceManager {
     // Announces |message| to all in-game administrators. Optionally |args| may be passed if
     // |message| is an instance of Message, which is common infrastructure for user-visible text.
     announceToAdministrators(message, ...args) {
+        this.announceToAdministratorsWithFilter(message, PlayerSetting.ANNOUNCEMENT.UNCATEGORIZED, PlayerSetting.SUBCOMMAND.GENERAL, args);
+    }
+
+    // Announces |message| to the administrators that have the announcements for |announceSubcategory| 
+    // and |subCommand| enabled. Optionally |args| may be passed if |message| is an instance of Message, 
+    // which is common infrastructure for user-visible text.
+    announceToAdministratorsWithFilter(message, announceSubcategory, subCommand, args) {
         if (message instanceof Message)
             message = Message.format(message, ...args);
 
         const formattedMessage = Message.format(Message.ANNOUNCE_ADMINISTRATORS, message);
 
+        var generalIdentifier = `${PlayerSetting.CATEGORY.ANNOUNCEMENT}/${announceSubcategory}/${PlayerSetting.SUBCOMMAND.GENERAL}`;
+        var specificIdentifier = `${PlayerSetting.CATEGORY.ANNOUNCEMENT}/${announceSubcategory}/${subCommand}`;
```

Comment:
Please do not use `var` anymore. It's got a ton of problems, particularly in regards of scoping. Consider the following:

```javascript
if (1 + 1 === 2) {
  var result = 3;
}

console.log(result);  // 3
```

Prefer `const` where you can, which is the case for all instances in this file. If you need to be able to reassign something to a variable, use `let`.

Example:

```javascript
const foo = 'bar';
foo = 'baz';  // this will throw an exception

let foo = 'bar'; 
foo = 'baz';  // this is fine
```

Reply:
Thank you, I'm totally aware of the differences and scoping.
As I mainly develop in C# var just is too common for me.